src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1007,18 +1007,48 @@ DQLiftGatherSymCW::DQLiftGatherSymCW() {
         auto matched_out_ids = uat::_(node_to_output).at_or_at(cvtids, pids);
         const auto& matched_out_gather = node_to_output.at(gather);
 
+        // Handle case with non-f16 scale
+        if (matched_out_s.get_element_type() != ov::element::f16) {
+            auto new_cvt_w = std::make_shared<ov::op::v0::Convert>(matched_out_w, ov::element::f16);
+            auto gather_c = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, 0);
+            auto new_g_w = std::make_shared<ov::op::v8::Gather>(new_cvt_w, matched_out_ids, gather_c);
+            auto new_cvt_s = std::make_shared<ov::op::v0::Convert>(matched_out_s, ov::element::f16);
+            auto new_g_s = std::make_shared<ov::op::v8::Gather>(new_cvt_s, matched_out_ids, gather_c);
+            auto new_mul = std::make_shared<ov::op::v1::Multiply>(new_g_w, new_g_s);
+            auto new_out = std::make_shared<ov::op::v0::Convert>(new_mul, matched_out_s.get_element_type());
+
+            NPUW_ASSERT(node_to_output.find(qcvtm) == node_to_output.end());
+
+            // Reconnect old gather readers to the new Convert
+            for (auto&& r : matched_out_gather.get_target_inputs()) {
+                r.replace_source_output(new_out);
+            }
+
+            return true;  // root was changed
+        }
+
         // Create new gathers on W and S, connect respectively
         auto new_cvt_w = std::make_shared<ov::op::v0::Convert>(matched_out_w, ov::element::f16);
         auto gather_c = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, 0);
         auto new_g_w = std::make_shared<ov::op::v8::Gather>(new_cvt_w, matched_out_ids, gather_c);
         auto new_g_s = std::make_shared<ov::op::v8::Gather>(matched_out_s, matched_out_ids, gather_c);
         auto new_mul = std::make_shared<ov::op::v1::Multiply>(new_g_w, new_g_s);
-        auto new_out = std::make_shared<ov::op::v0::Convert>(new_mul, ov::element::f32);
+        std::shared_ptr<ov::Node> new_out = new_mul;
+        if (matched_out_s.get_element_type() != ov::element::f16) {
```

Comment:
this code is unreachable, isn't it? in the line 1011 you get that if already and make an early return